src/model/ScheduleDay.cpp

```diff
@@ -176,42 +181,123 @@ namespace model {
         return 0.0;
       }
 
-      std::vector<double> values = this->values();          // these are already sorted
-      std::vector<openstudio::Time> times = this->times();  // these are already sorted
+      TimeSeries ts = this->timeSeries();
 
-      unsigned N = times.size();
+      DateTimeVector dateTimes = ts.dateTimes();
+      Vector values = ts.values();
+
+      unsigned N = dateTimes.size();
```

Comment:
Mmh, starting there you lost me. I hadn't seen this bit when I wrote the comment above, and I hadn't though about a time being requested in between actual timesteps, but this looks wrong to me.

Let me see if I follow:

* You get timeSeries, which is already something interpolated, but aligns on the Timeseries::timestep
* Then you scan the **entire** day for that, and you call another interpolation call

I will go look into Vector/Matrix next, but this doesn't sit right with me.

Reply:
The second one (i.e., getValue) is not _really_ an interpolation call. It is a call with `HoldNextInterp`.

Reply:
So, I'm really still not a big fan of this. But I don't think it warrants blocking the PR... I expect most people will just use timeSeries anyways.